feat: 브랜드 색상 시스템 및 GlobalLayout 컴포넌트 구현

woo-dev-log opened the pull request:
### 📌 PR 종류 (PR Type)
<!-- 아래 항목 중 해당하는 곳에 [x] 로 표시해주세요. -->

- [ ] 🐛 버그 수정 (Bugfix)
- [x] ✨ 기능 추가 (Feature)
- [x] 🎨 UI 스타일 변경 (Style) - CSS, 클래스명, 레이아웃 등 UI 변경
- [ ] ♻️ 코드 리팩토링 (Refactor) - 기능 변화 없이 코드 구조 개선
- [ ] 📦 패키지/빌드 관련 변경 (Build) - 설정 파일 수정, 의존성 추가 등
- [ ] 🧱 테스트 코드 추가/수정 (Test)
- [ ] 📝 문서 수정 (Docs)
- [ ] 🚧 WIP (작업 중 PR)
- [ ] 기타 (Other): <!-- 직접 입력 -->

---

### 📝 요약 (Summary)
<!-- 한두 문장으로 변경 내용을 요약해주세요. -->
브랜드 색상 시스템 구축 및 480px 고정 모바일 레이아웃 GlobalLayout 컴포넌트 구현

---

### 🔍 상세 내용 (Describe your changes)
<!-- 어떤 변경사항이 있었는지, 왜 변경했는지 상세히 작성해주세요. 코드 흐름, 주요 로직 등을 설명해도 좋습니다. -->
- Tailwind CSS v4의 `@theme` 블록을 활용하여 브랜드 색상 4개 정의
- `*-brand-*` 클래스로 사용 가능한 색상 시스템 구축

**정의된 브랜드 색상**
- `brand-background`: #1B1B1B (메인 배경색)
- `brand-primary`: #99FF4A (주요 브랜드 컬러 - 라임 그린)
- `brand-surface`: #2C2C2C (카드/보드 표면색)
- `brand-text`: #7E7E7E (텍스트 색상)

#### GlobalLayout 컴포넌트 구현
- 480px 고정 너비로 모바일 최적화된 레이아웃 구조 설계
- 외부 영역 slate-100 배경으로 모바일 앱 시뮬레이션 효과 구현
- 헤더(56px), 메인, 푸터(56px) 3단 구조로 확장 가능한 레이아웃 설계
- 메인 콘텐츠 영역 flex-grow로 동적 높이 조절 및 16px 패딩 적용

#### 기술적 구현
- Tailwind CSS v4 네이티브 방식으로 색상 관리 (별도 config 파일 불필요)
- shadcn/ui 디자인 시스템과 완벽 호환

### 📸 스크린샷 (필수)
<!-- UI 변경이 있는 경우, 변경 전/후 비교나 주요 화면을 첨부해주세요. -->
| Before | After |
|--------|-------|
|        |<img width="1460" height="952" alt="image" src="https://github.com/user-attachments/assets/c5b683dd-1413-441e-b197-c890fba55a2b" />|

---

### ✅ 확인 사항 (CheckList)
<!-- PR을 보내기 전 스스로 점검할 수 있도록 합니다. 팀원이 리뷰할 때도 참고할 수 있습니다. -->
- [x] 브라우저에서 정상 동작 확인
- [x] 기존 기능에 영향 없음
- [x] 반응형 UI 대응 확인
- [x] 코드 컨벤션에 맞게 작성됨 (eslint, prettier 등)
- [x] 테스트 코드 추가/수정 또는 테스트 통과 확인

**추가 확인 사항**
- [x] 브랜드 색상 클래스 정상 작동 확인 (`bg-brand-primary`, `text-brand-text` 등)
- [x] 다양한 화면 크기에서 레이아웃 중앙 정렬 확인

---
### 💬 기타 참고사항 (Optional)
<!-- 리뷰어가 알면 좋을 추가 정보, 요청 사항 등을 작성해주세요. -->
#### 리뷰 포인트
- 브랜드 색상 선택과 네이밍이 적절한지 확인 부탁드립니다.
- 현재 구조가 적합한지 의견 부탁드립니다.

#### 색상 관리 방식 선택에 대한 검토 요청
`color.ts` 대신 `@theme` 블록을 선택한 이유
- Tailwind CSS v4 네이티브 방식: 별도 JavaScript 설정 없이 CSS만으로 색상 관리 가능
- 개발 속도: 빠른 프로토타이핑을 위해 설정이 간단한 방식 선택
- 성능: 런타임 오버헤드 없이 컴파일 타임에 색상 처리
- 동적 테마 변경 같은 다른 기능이 필요하다면 `color.ts` 방식으로 전환 가능

#### 참고 자료
- [Tailwind CSS v4 Theme Configuration](https://tailwindcss.com/docs/theme)

woo-dev-log committed:
feat: 브랜드 색상 및 GlobalLayout 구현

- Tailwind v4 @theme 블록에 브랜드 색상 정의
- 480px 고정 모바일 레이아웃 구현
- 헤더/메인/푸터 3단 구조 설정

woo-dev-log committed:
Merge branch 'develop' into feature/global-layout

ksy90101 reviewed (approved):
👍

Monixc commented on src/index.css:
```diff
@@ -3,6 +3,13 @@
+  --color-brand-surface: #2C2C2C;
+  --color-brand-text: #7E7E7E;
+}
+
```
brand 컬러 네이밍은 너무 좋습니다! 역할별로 잘 정리되어 있는 것 같아요👍

제가 아까 요청드릴 땐 HEX값으로 드리긴 했지만, 지금 보니 커스텀 테마를 사용하면서 Tailwind 기본 설정이 유지되고 있는 구조라, 색상 시스템은 `oklch()`로 통일하는 편이 더 일관성 있을 것 같습니다.

`oklch()`가 시각적으로 더 자연스러운 명도/대비 조절 가능하고, 또 다크 모드에서 색상 균형 잡기에도 유리하다고 하네요. 사실 지금은 사용하는 컬러가 4개 정도로 단순해서 HEX로도 충분하긴 한데, 색상 시스템을 확장하게 될 가능성을 생각하면 미리 `oklch()`로 통일해 두는 것도 괜찮을 것 같아서요!

아래는 기존 HEX 컬러를 oklch()로 변환한 값인데, 이대로 반영해주실 수 있을까요??

```
@theme {
  --color-brand-background: oklch(0.2221 0 0);       /* #1B1B1B */
  --color-brand-primary: oklch(0.9042 0.2304 134.57); /* #99FF4A */
  --color-brand-surface: oklch(0.2931 0 0);          /* #2C2C2C */
  --color-brand-text: oklch(0.5931 0 0);             /* #7E7E7E */
}
```

참고자료: 
- https://www.linkedin.com/posts/vitalyfriedman_colors-design-css-activity-7062428890362699776-LdLe/
- https://chamika-karunarathna.medium.com/oklch-the-future-of-color-in-web-development-93f0c21ed8fc
- 변환 사이트: https://oklch.com/#0.5931,0,0,100

woo-dev-log committed:
Merge branch 'develop' into feature/global-layout

woo-dev-log committed:
style: HEX 색상을 OKLCH로 변환하여 색상 시스템 일관성 개선

- 브랜드 색상 4개를 HEX에서 OKLCH 형식으로 변환
- shadcn/ui 기본 색상 시스템과 통일성 확보
- 향후 색상 시스템 확장 시 일관성 유지
